ttnn/cpp/ttnn/operations/pool/grid_sample/device/grid_sample_nearest_program_factory.cpp

```diff
@@ -0,0 +1,271 @@
+    if (enable_split_reader && !is_sharded) {
+        std::tie(grid_cb_index1, grid_cb_handle1) =
+            tt::tt_metal::create_cb(cb_idx++, program, all_cores, grid_stick_size, 1, grid_cb_data_format, nullptr);
+    }
+
+    const uint32_t output_cb_page_size = (float)output_shape[-1] * output_tensor.element_size();
+    const uint32_t output_cb_pages = output_nsticks_per_core;
+    const auto [output_cb_index, output_cb_handle] = tt::tt_metal::create_cb(
+        cb_idx++,
+        program,
+        all_cores,
+        output_cb_page_size,
+        output_cb_pages,
+        output_cb_data_format,
+        output_tensor.buffer());
+
+    // Prepare stick size arguments with proper names
+    const uint32_t input_stick_size = get_aligned_stick_size(input_shape, input_tensor);
+    const uint32_t grid_stick_size_arg =
+        is_sharded ? grid_shape[-1] * grid_tensor.element_size() : get_aligned_stick_size(grid_shape, grid_tensor);
+
+    // Writer kernel
+    tt::tt_metal::KernelHandle writer_kernel_id = 0, writer1_kernel_id = 0;
+
+    // Writer for nearest mode with sharded grid - matches sharded reader argument structure
+    std::vector<uint32_t> writer_compile_time_args = {
+        grid_cb_index0,                              // ct_arg[0]: grid_cb_index
+        output_cb_index,                             // ct_arg[1]: output_cb_index (replaces scalar_cb_index)
+        input_stick_size,                            // ct_arg[2]: input_stick_size
+        grid_stick_size_arg,                         // ct_arg[3]: grid_stick_size
+        input_height,                                // ct_arg[4]: input_height
+        input_width,                                 // ct_arg[5]: input_width
+        grid_batching_factor,                        // ct_arg[6]: grid_batching_factor
+        static_cast<uint32_t>(grid_tensor.dtype()),  // ct_arg[7]: grid_dtype
+        grid_hw,                                     // ct_arg[8]: grid_hw
+        use_precomputed_grid ? 1U : 0U,              // ct_arg[9]: use_precomputed_grid
+        align_corners ? 1U : 0U,                     // ct_arg[10]: align_corners
+        enable_split_reader ? 1U : 0U,               // ct_arg[11]: split_reader (same as reader)
+        0U,                                          // ct_arg[12]: reader_id (will be set per core)
+        grid_nsticks_per_core,                       // ct_arg[13]: grid_nsticks_per_core
+        is_sharded ? 1U : 0U                         // ct_arg[14]: is_sharded
+    };
+
+    // Add tensor accessor args for input tensor (15 compile time args offset)
+    tt::tt_metal::TensorAccessorArgs(*input_tensor.buffer()).append_to(writer_compile_time_args);
+    if (!is_sharded) {
+        tt::tt_metal::TensorAccessorArgs(*grid_tensor.buffer()).append_to(writer_compile_time_args);
+    } else {
+        tt::tt_metal::TensorAccessorArgs(*input_tensor.buffer()).append_to(writer_compile_time_args);
```

Comment:
what is the idea here for tensor accessor in case is_sharded is true, why would we pass infor for input tensor twice?